packages/app/src/cli/services/flow/validation.ts

```diff
@@ -25,6 +25,28 @@ export const validateNonCommerceObjectShape = (configField: ConfigField, type: F
     }
   }
 
+  if (Object.keys(SUPPORTED_COMMERCE_OBJECTS).includes(configField.type)) {
+    if (configField.key) {
+      throw new zod.ZodError([
+        {
+          code: zod.ZodIssueCode.custom,
+          path: ['settings.fields.key'],
+          message: 'Key is not a valid property for commerce object fields',
+        },
+      ])
+    }
+
+    if (configField.name && type === 'flow_action') {
+      throw new zod.ZodError([
+        {
+          code: zod.ZodIssueCode.custom,
+          path: ['settings.fields.name'],
+          message: 'Name is not a valid property for commerce object fields',
+        },
+      ])
+    }
+  }
```

Comment:
instead of having to validate each properties when the devs use a commerce object, I feel it's better to have a zod schema for it at this point. It's a nice to have though.

```javascript
import { z } from "zod";

const FlowActionCommerceField = z.object({
  type: z.nativeEnum(SUPPORTED_COMMERCE_OBJECTS as const),
  description: z.string().optional(),
  required: z.boolean(),
});

FlowActionCommerceField.parse(configField);

```

Reply:
Good idea!

Reply:
Yes! Let's do this please, zod all the validations 🙏

Reply:
You [can](https://zod.dev/?id=strings) customize all `zod` error messages: 
```ts
type: z.nativeEnum(SUPPORTED_COMMERCE_OBJECTS as const, {message: 'type must be....'}),
```